src/instructlab/model/dk_bench_utils.py

```diff
@@ -0,0 +1,310 @@
+
+
+class IOFileType(enum.Enum):
+    CSV = "csv"
+    JSONL = "jsonl"
+    XLSX = "xlsx"
+    QNA_YAML = "qna.yaml"
+
+
+def model_is_localand_valid(model: str) -> bool:
+    valid_model = False
+    if model is not None and os.path.exists(model):
+        model_path = pathlib.Path(model)
+        if model_path.is_dir():
+            valid_model = is_model_safetensors(model_path)
+        elif model_path.is_file():
+            valid_model = is_model_gguf(model_path)
+    return valid_model
+
+
+def get_endpoint_model_name(endpoint: str, api_key: str | None, client: httpx.Client) -> str:
+    try:
+        openai_client = OpenAI(
+            base_url=endpoint,
+            api_key=api_key,
+            timeout=DEFAULTS.CONNECTION_TIMEOUT,
+            http_client=client,
+        )
+        models = openai_client.models.list()
+    except OpenAIError as exc:
+        raise ClientException(f"Connection Error {exc}") from exc
+
+    if len(models.data) != 1:
+        raise ClientException(
+            "More than one model at endpoint, pass in model name manually"
+        )
+
+    return models.data[0].id
+
+
+def create_results_file_name(
+    file_format: str, output_dir: str, timestamp: str, model_name: str
+) -> str:
+    if IOFileType.CSV.value == file_format:
+        file_type = "csv"
+    elif IOFileType.XLSX.value == file_format:
+        file_type = "xlsx"
+    elif IOFileType.JSONL.value == file_format:
+        file_type = "jsonl"
+    else:
+        raise ValueError("File format is not one of: csv, xlsx, jsonl")
+
+    output_path = pathlib.Path(output_dir)
+    if not output_path.exists() or not output_path.is_dir():
+        output_dir = DEFAULTS.EVAL_DATA_DIR
+        logger.debug(
+            "Output dir provided is not a directory or does not exist, writing results to default evaluation output directory %s",
+            output_dir,
+        )
+
+    return f"{output_dir}/responses-{model_name}-{timestamp}.{file_type}"
+
+
+def get_api_base_for_model(
+    ctx: click.Context,
+    model: str,
+    temperature: float,
+    model_prompt: str,
+    max_workers: str | int | None,
+    gpus: int | None,
+    backend: str | None,
+    enable_serving_output: bool,
+) -> tuple:
+    server = None
+    if model_is_localand_valid(model):
+        logger.debug("DK-Bench model is local")
+        model_name = get_model_name(model)
+        server, api_base, _ = launch_server(
+            eval_serve=ctx.obj.config.serve,
+            tls_client_cert=ctx.params["tls_client_cert"],
+            tls_client_key=ctx.params["tls_client_key"],
+            tls_client_passwd=ctx.params["tls_client_passwd"],
+            tls_insecure=ctx.params["tls_insecure"],
+            model=model,
+            model_name=model_name,
+            max_workers=max_workers,
+            gpus=gpus,
+            backend=backend,
+            enable_serving_output=enable_serving_output,
+        )
+    else:
+        logger.debug("DK-Bench model is an endpoint")
```

Comment:
Why are we calling it a `dk-bench` model? How is this different from a student model?

Reply:
It's the same thing, but I'll rephrase this to `Model being evaluated in DK-Bench is an endpoint`.